android/src/com/frostwire/android/core/FileDescriptor.java

```diff
@@ -82,6 +84,39 @@ public FileDescriptor(int id, String artist, String title, String album, String
         ensureCorrectMimeType(this);
     }
 
+    public FileDescriptor(Bundle bundle) {
+        this(bundle.getInt("id"),
```

Comment:
this is interesting, since none of the fields is `final`, I suppose that for consistency, it should be a `fromBundle` and use that here instead.